queries.sql

```diff
@@ -22,4 +22,40 @@ SELECT * FROM animals WHERE neutered;
 SELECT * FROM animals WHERE name not like 'Gabumon';
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 and 17.3; 
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 and 17.3; 
+
+/*  Transactions  */
+
+/* Stage 1 */
+BEGIN;
+UPDATE animals SET species='unspecified';
+ROLLBACK;
+
+/* Stage 2 */
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species='pokemon' WHERE name NOT LIKE '%mon';
+COMMIT;
+
+/* Stage 3 DELETE */ 
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+/* Stage 4 */ 
+BEGIN;
+DELETE FROM animals WHERE date_of_birth >= '2022-01-01';
+SAVEPOINT firstpoint;
+UPDATE animals SET weight_kg = weight_kg*-1;
+ROLLBACK TO firstpoint;
+
+/* Stage 5 */ 
+UPDATE animals SET weight_kg = weight_kg*-1 WHERE weight_kg<0;
+COMMIT;
+
+/* Queries */
+SELECT COUNT( name ) FROM animals;
+SELECT COUNT( * ) FROM animals WHERE escape_attempts = 0;
+SELECT AVG( weight_kg ) FROM animals;
+SELECT neutered, AVG( escape_attempts ) FROM animals GROUP BY neutered ORDER BY AVG DESC LIMIT 1;
```

Comment:
- [ ] I think here you don't need to add ordered by avg since the question is just `Who escapes the most, neutered or not neutered animals?` so it just we want to make it Group by neutered  so kindly remove it 💪🏻😊🚀